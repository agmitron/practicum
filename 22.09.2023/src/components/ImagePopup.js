import React from "react";
import PropTypes from 'prop-types';
import Popup from "./Popup.js";

function ImagePopup({ card, onClose }) {
  const isOpen = !!card;

  return (
    <Popup
      isOpen={isOpen}
      onClose={onClose}
      popupClass='popup_type_image'
      contentClass='popup__content_content_image'
    >
      <img
        alt={card ? card.name : ""}
        src={card ? card.link : ""}
        className='popup__image'
      />
      <p className='popup__caption'>{card ? card.name : ""}</p>
    </Popup>
  );
}

ImagePopup.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  }),
  onClose: PropTypes.func.isRequired
}

export default ImagePopup;
