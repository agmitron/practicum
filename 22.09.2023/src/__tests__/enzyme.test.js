import ImagePopup from '../components/ImagePopup';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

let mockCard = {
    "likes": [],
    "_id": "1",
    "name": "Deer",
    "link": "https://cat.photo",
    "owner":
    {
        "name": "John",
        "about": "Web-developer",
        "avatar": "https://avatar.photo",
        "_id": "1",
        "cohort": "cohort-13"
    },
    "createdAt": "2021-10-24T19:41:51.239Z"
};

let mockCard_cat = {
    "likes": [],
    "_id": "1",
    "name": "Cat",
    "link": "https://deer.photo",
    "owner":
    {
        "name": "John",
        "about": "Web-developer",
        "avatar": "https://avatar.photo",
        "_id": "1",
        "cohort": "cohort-13"
    },
    "createdAt": "2021-10-24T19:41:51.239Z"
};

const clickFn = jest.fn();
describe('>>>H O M E --- Shallow Render REACT COMPONENTS',()=>{

    Enzyme.configure({ adapter: new Adapter() });
    let wrapper

    beforeEach(()=>{
        wrapper = Enzyme.shallow(<ImagePopup card={mockCard} onClose={clickFn}/>)
    })

    test('+++ render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    test('+++ render the DUMB snap', () => {
        expect(wrapper).toMatchSnapshot();
     });
      
    test('+++ contains output', () => {
        wrapper
            .find('Popup')
            .simulate('close');
        expect(clickFn).toHaveBeenCalled();
    });

    test('+++ contains app', () => {
        const caption = wrapper.find('.popup__caption');
        const result = caption.text();
        expect(result).toEqual('Deer');
    });

    test('+++ new props app', () => {
        wrapper.setProps({ card: mockCard_cat });
        const caption = wrapper.find('.popup__caption');
        const result = caption.text();
        expect(result).toEqual('Cat');
    });
});