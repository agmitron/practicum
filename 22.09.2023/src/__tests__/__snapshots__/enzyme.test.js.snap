// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>H O M E --- Shallow Render REACT COMPONENTS +++ render the DUMB snap 1`] = `
<Popup
  contentClass="popup__content_content_image"
  isOpen={true}
  onClose={[MockFunction]}
  popupClass="popup_type_image"
>
  <img
    alt="Deer"
    className="popup__image"
    src="https://cat.photo"
  />
  <p
    className="popup__caption"
  >
    Deer
  </p>
</Popup>
`;
